safehttp/form.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"errors"
+	"mime/multipart"
+	"strconv"
+)
+
+// Form TODO
+type Form struct {
+	values map[string][]string
+	err    error
+	parsed bool
+}
+
+// MultipartForm TODO
+type MultipartForm struct {
+	form Form
+	file map[string][]*multipart.FileHeader
+}
```

Comment:
Keep the definition of a struct with it's method's definition together in the file.

Reply:
As discussed, I used embedding in[542ad4d](url) , so I left the two types in the same file.